olp-cpp-sdk-core/src/http/curl/NetworkCurl.cpp

```diff
@@ -871,59 +850,52 @@ size_t NetworkCurl::HeaderFunction(char* ptr, size_t size, size_t nitems,
 
 void NetworkCurl::CompleteMessage(CURL* handle, CURLcode result) {
   std::unique_lock<std::mutex> lock(event_mutex_);
-  size_t index = 0;
-  for (; index < handles_.size(); ++index) {
-    if (handles_[index].in_use && (handles_[index].handle == handle)) {
-      break;
-    }
-  }
+  int index = GetHandleIndex(handle);
+  if (index >= 0 && index < static_cast<int>(handles_.size())) {
+    RequestHandle& rhandle = handles_[index];
```

Comment:
Usually, abbreviations are not allowed but we can close one eye for this.
Else i would suggest to name it `request_handle` or just `handle`.